letsencrypt/error_handler.py

```diff
@@ -0,0 +1,59 @@
+"""Registers and calls cleanup functions in case of an error."""
+import os
+import signal
+
+
+_SIGNALS = ([signal.SIGTERM] if os.name == "nt" else
+            [signal.SIGTERM, signal.SIGHUP, signal.SIGQUIT,
+             signal.SIGXCPU, signal.SIGXFSZ, signal.SIGPWR])
+
+
+class ErrorHandler(object):
+    """Registers and calls cleanup functions in case of an error."""
```

Comment:
It should be documented what kind of errors are covered: i.e. Python exceptions **and** signals. (it's clear from PR description, but not from the code). I think it should be explicitly stated that this does not catch exceptions and that previously registered signal handler is still called. Otherwise one has to get into the fine details in the source code in order to understand provided API.

I would also document typical usage scenario (i.e. that this is actually meant to work as context manager). Maybe give an example code?